### PR TITLE
add cluster-api & controller-manager builders

### PR DESCRIPTION
**Pull request by paulfantom**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,13 +1,61 @@
...
 
-GO_BUILD= env GOOS=$(PLATFORM) GOARCH=$(ARCH) go build -i $(GOFLAGS) \
+# Define constants
+##################
+BINDIR        ?= bin
+PLATFORM      ?= linux
+ARCH          ?= amd64
+CLUSTERAPI_BIN = $(BINDIR)/cluster-api
+VERSION       ?= $(shell git describe --always --abbrev=7 --dirty)
+GO_VERSION     = 1.9
```
Comment (enxebre) [inline Makefile]: Any reason why we can't default to 1.10

File: Makefile [hunk 2]
```diff
@@ -1,13 +1,61 @@
...
+build: kubernetes-cluster-api kubernetes-controller-manager aws-machine-controller
 
 .PHONY: $(BINDIR)/aws-machine-controller
 aws-machine-controller: $(BINDIR)/aws-machine-controller
 $(BINDIR)/aws-machine-controller:
 	$(GO_BUILD) -o $@ ./cmd
 
 aws-machine-controller-image: $(BINDIR)/aws-machine-controller
-	docker build -t aws-machine-controller --file build/aws-machine-controller/Dockerfile .
+	docker build -t $(AWS_MACHINE_CONTROLLER_IMAGE) --file build/aws-machine-controller/Dockerfile .
```
Comment (enxebre) [inline Makefile]: can we align this with the way we build the other images? either always use --file or always use cp

File: Makefile [hunk 3]
```diff
@@ -1,13 +1,61 @@
...
+
+.PHONY: $(CLUSTERAPI_BIN)/apiserver
+$(CLUSTERAPI_BIN)/apiserver: .apiServerBuilderImage
+	mkdir -p $(PWD)/$(CLUSTERAPI_BIN) && docker run --security-opt label:disable -v $(PWD)/$(CLUSTERAPI_BIN):/output --entrypoint=/bin/bash apiserverbuilderimage -c "export GOPATH=/go && mkdir -p /go/src/sigs.k8s.io/cluster-api && cd /go/src/sigs.k8s.io/cluster-api && git clone https://github.com/kubernetes-sigs/cluster-api.git . && apiserver-boot build executables --generate=false && touch /output/controller-manager /output/apiserver && cp bin/* /output"
+
+.PHONY: kubernetes-cluster-api
+kubernetes-cluster-api: $(CLUSTERAPI_BIN)/apiserver build/clusterapi-image/Dockerfile
```
Comment (enxebre) [inline Makefile]: can we keep naming consistent so  if we keep aws-machine-controller-image, we should add image here as well

File: Makefile (hunk 2 above)
Comment (paulfantom) [reply Makefile]: Ok, I'll use cp so it will be the same way as in cluster-operator.

File: Makefile (hunk 1 above)
Comment (paulfantom) [reply Makefile]: It was a copy-paste from cluster-operator

Comment (enxebre) [reply Makefile]: I know, so the question still stands :)

Comment (paulfantom) [reply Makefile]: I don't see any reason.

File: Makefile [hunk 4]
```diff
@@ -1,13 +1,83 @@
...
+
+
+CLUSTER_API_IMAGE = $(REGISTRY)cluster-api:$(VERSION)
+CONTROLLER_MANAGER_IMAGE = $(REGISTRY)controller-manager:$(VERSION)
+AWS_MACHINE_CONTROLLER_IMAGE = $(REGISTRY)aws-machine-controller:$(VERSION)
+
+# Some prereq stuff
+###################
+
+.clusterOperatorBuildImage: build/build-image/Dockerfile
```
Comment (enxebre) [inline Makefile]: this seems unnecessary reference to cluster operator form now on

**Comment by enxebre** [comment]
let's keep it as simple as possible for now. lgtm thanks!

**Pull request closed without merging**